✨ Source Facebook Marketing, Salesforce, Shopify, Stripe: add maxSecondsBetweenMessages in metadata

strosek opened the pull request:
Certified connectors are required to have `maxSecondsBetweenMessages` set in metadata file.
`maxSecondsBetweenMessages` is the longest time frame (in seconds) of API requests limits reset for endpoints used by connector. [Issue](https://app.zenhub.com/workspaces/critical-connectors-659c693d906fcd00178900cf/issues/gh/airbytehq/airbyte-internal-issues/6779)

## What
`maxSecondsBetweenMessages` value for the following connectors was added:

source-stripe [reference](https://docs.stripe.com/rate-limits)
source-facebook-marketing [reference](https://developers.facebook.com/docs/marketing-apis/rate-limiting)
source-salesforce [reference](https://developer.salesforce.com/docs/atlas.en-us.salesforce_app_limits_cheatsheet.meta/salesforce_app_limits_cheatsheet/salesforce_app_limits_platform_api.htm)
source-shopify [reference](https://shopify.dev/docs/api/usage/rate-limits)

## How
Files `metadata.yaml` for connectors mentioned above were updated by adding `maxSecondsBetweenMessages` value.

## 🚨 User Impact 🚨
No breaking changes

Commit:
Add maxSecondsBetweenMessages for Salesforce connector

Commit:
Add maxSecondsBetweenMessages for Salesforce connector

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/airbytehq/airbyte?pullRequest=36472) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/airbytehq/airbyte?pullRequest=36472) before we can accept your contribution.<br/>**28** out of **29** committers have signed the CLA.<br/><br/>:white_check_mark: girarda<br/>:white_check_mark: gisripa<br/>:white_check_mark: erohmensing<br/>:white_check_mark: aaronsteers<br/>:white_check_mark: JonsSpaghetti<br/>:white_check_mark: natikgadzhi<br/>:white_check_mark: edgao<br/>:white_check_mark: DanyloGL<br/>:white_check_mark: pnilan<br/>:white_check_mark: tolik0<br/>:white_check_mark: davydov-d<br/>:white_check_mark: stephane-airbyte<br/>:white_check_mark: nataliekwong<br/>:white_check_mark: ChristoGrab<br/>:white_check_mark: artem1205<br/>:white_check_mark: akashkulk<br/>:white_check_mark: cjkindel<br/>:white_check_mark: brianjlai<br/>:white_check_mark: Abner1zhou<br/>:white_check_mark: nobowned<br/>:white_check_mark: avirajsingh7<br/>:white_check_mark: darynaishchenko<br/>:white_check_mark: graingert-coef<br/>:white_check_mark: gosusnp<br/>:white_check_mark: henriblancke<br/>:white_check_mark: roman-yermilov-gl<br/>:white_check_mark: lazebnyi<br/>:white_check_mark: bengineerdavis<br/>:x: Erick Corona<br/><hr/>**Erick Corona** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/airbytehq/airbyte?pullRequest=36472) it.</sub>

girarda committed:
🤖 Bump minor version of Python CDK

gisripa committed:
Destinations cdk: Cleanup of obsolete interfaces (#36465)

natikgadzhi committed:
airbyte-ci README cleanup part 2 (#36426)

## What

Following up and adding more structure to airbyte-ci readme. No code changes. Following up #36239.

## Follow-up work

- [ ] @alafanechere mentioned before — we should clean out `connector_ops` and move the pieces we need into `common_utils`, remove the package.
- [ ] Rename `connectors_qa` into `connector_metadata_checks`
- [ ] Clarify what exactly `qa-engine` is doing.

maxi297 reviewed (approved):
LGTM! Conditional approval on @davinchia 's approval.

A couple of things regarding the documentation of PR:
* Can you link the ticket here? We have multiple repos with issues so having the link will help relate to the actual issues 
* Can we update the ticket based on [those comments](https://airbytehq-team.slack.com/archives/C06H4LJRK1Q/p1711379371421849?thread_ts=1711377477.240559&cid=C06H4LJRK1Q) to explain why we're not doing the others in the ticket (yet)

erohmensing committed:
Airbyte CDK: move from gradle to poetry (#36250)

erohmensing committed:
🤖 Bump minor version of Python CDK

girarda committed:
low-code: Add string filter  (#36393)

Co-authored-by: Marcos Marx <marcosmarxm@users.noreply.github.com>

girarda committed:
🤖 Bump minor version of Python CDK

aaronsteers committed:
Fix Vector destination tests and move to Poetry (#35911)

Co-authored-by: bindipankhudi <bindi@airbyte.com>
Co-authored-by: bindipankhudi <bindi@airbyte.io>

JonsSpaghetti committed:
Bump Airbyte version from 0.55.1 to 0.55.2

girarda committed:
Add low-code filters to the reference docs (#36469)

Co-authored-by: Lake Mossman <lake@airbyte.io>

Commit:
DBZ iterator migration to use SourceStateIterator (#36333)

DanyloGL committed:
Set `maxSecondsBetweenMessages` value for `certified` connectors Phase 1 (#36382)

tolik0 committed:
:bug: Source Chargebee/Monday: Disable state validation test (#36380)

Co-authored-by: Roman Yermilov <roman.yermilov@globallogic.com>

erohmensing committed:
CDK: fix empty extras (#36485)

erohmensing committed:
🤖 Bump patch version of Python CDK

erohmensing committed:
Don't try to run connectors CI tests if only the python CDK changed (#36489)

Save 6 precious minutes every time we change the Python CDK and nothing else :)

edgao committed:
Destination snowflake: Depend on published cdk artifact (#36461)

Co-authored-by: alafanechere <augustin.lafanechere@gmail.com>

pnilan committed:
airbyte-ci: add `htmlcov` to list of ignored directories for security check (#36359)

akashkulk committed:
[Source-postgres] : Source operations suport for meta column  (#36432)

erohmensing committed:
Fix E721 errors in the CDK (#36490)

tolik0 committed:
🎉 Source Freshdesk: Migrate to low-code (#35699)

davydov-d committed:
:bug: Source Github: increase discovery time (#36429)

edgao committed:
Destination snowflake: more fixes for case-sensitivity (#36466)

stephane-airbyte committed:
force-publish java CDK on push to master (#36476)

maxi297 commented:
I'm fine with Davin not approving following our conversations. Feel free to merge any time you are comfortable with!

tolik0 committed:
:sparkles: Source GCS: Add Gzip and Bzip compression support (#36373)

Co-authored-by: Artem Inzhyyants <artem.inzhyyants@gmail.com>

nataliekwong committed:
[Docs] Simplify Connection Settings (#36502)

Co-authored-by: Chandler Prall <chandler.prall@gmail.com>

ChristoGrab committed:
Source Sendgrid: unpin CDK version (#36455)

stephane-airbyte committed:
bump java CDK version to 0.28.0 (#36514)

stephane-airbyte committed:
core java CDK core/testFixtures to kotlin (#36190)

This is an automated translation by intellij

Commit:
Add facebook-marketing, shopify and stripe constants

Commit:
Add facebook-marketing, shopify and stripe constants

stephane-airbyte committed:
convert java CDK core/test to kotlin (#36204)

stephane-airbyte committed:
convert java CDK core/main to kotlin (#36205)

This is an automated translation by intellij

stephane-airbyte committed:
convert datastore-bigquery to kotlin (#36411)

convert datastore-bigquery to kotlin

fix compilation errors

stephane-airbyte committed:
convert java-cdk/mongodb to kotlin (#36412)

convert java-cdk/mongodb to kotlin

fix compilation errors

stephane-airbyte committed:
convert java CDK postgres submodule to kotlin (#36413)

convert java CDK postgres submodule to kotlin

fix compilation errors

stephane-airbyte committed:
convert cdk azure submodule to kotlin (#36419)

convert cdk azure submodule to kotlin

fix compiler errors

stephane-airbyte committed:
convert CDK s3-destinations to kotlin (#36420)

convert CDK s3-destinations to kotlin

fix compiler errors

stephane-airbyte committed:
convert typing_deduping to kotlin (#36421)

convert typing_deduping to kotlin

fix compiler errors

stephane-airbyte committed:
convert CDK gsc submodule to kotlin (#36437)

convert CDK gsc submodule to kotlin

fix compiler errors

erohmensing committed:
Update out of date internal lockfiles (#36535)

artem1205 committed:
Source Azure Blob Storage: migrate to poetry (#36487)

Signed-off-by: Artem Inzhyyants <artem.inzhyyants@gmail.com>
Co-authored-by: Catherine Noll <noll.catherine@gmail.com>

bengineerdavis committed:
Bengineer/fix updating documentation local dev requirements (#36522)

stephane-airbyte committed:
convert CDK db-sources submodules to kotlin (#36438)

convert CDK db-sources submodules to kotlin

fix compiler warnings

stephane-airbyte committed:
convert java CDK db-destinations submodule to kotlin (#36439)

convert java CDK db-destinations submodule to kotlin

fix compiler warnings

stephane-airbyte committed:
convert #36396 to kotlin (#36473)

stephane-airbyte committed:
bring in #36465 (#36467)

stephane-airbyte committed:
use appache commons classes in our Checked functional interfaces (#36444)

stephane-airbyte committed:
convert functional interfaces to kotlin (#36445)

erohmensing committed:
delete unused cdk scripts (#36496)

erohmensing committed:
airbyte ci test to support --extras (#36527)

Our `airbyte-ci test` command supports [groups](https://python-poetry.org/docs/pyproject/#dependencies-and-dependency-groups) but not [extras](https://python-poetry.org/docs/pyproject/#extras). This PR allows configuring extras to install if necessary (like when running the unit tests in the [upstack PR](https://github.com/airbytehq/airbyte/actions/runs/8446603559/job/23135651978?pr=36497)), and renames to better distinguish between the groups. 

We probably don't need to include `dev` as an optional group since it's implicitly installed, but I didn't mess with that here, for one because I wanted to test the rename.

For usage see [upstack PR.](https://github.com/airbytehq/airbyte/pull/36497/files#diff-1734d8b375bd2fc6fd8419f94ca88aad97eefaf2e78fa548c7379b28fca6456dR97-R101)

Abner1zhou committed:
📝 [docs] update instructions to use new connectors (#36298)

Co-authored-by: Marcos Marx <marcosmarxm@users.noreply.github.com>

erohmensing committed:
check cdk: don't use connector test runenr  (#36564)

erohmensing committed:
airbyte cdk: simplify datamodel-gen scripts (#36530)

brianjlai committed:
[airbyte-cdk] emit source recordCount as float instead of integer (#36560)

Co-authored-by: Ella Rohm-Ensing <erohmensing@gmail.com>

brianjlai committed:
🤖 Bump patch version of Python CDK

stephane-airbyte committed:
convert java CDK dependencies submodule to kotlin (#36446)

convert java CDK dependencies submodule to kotlin

fix compiler errors

stephane-airbyte committed:
convert missing s3-destinations tests (#36570)

cjkindel committed:
🐛[source-confluence] 'expand' request parameter converted to comma separated string (#36339)

Co-authored-by: Marcos Marx <marcosmarxm@users.noreply.github.com>
Co-authored-by: marcosmarxm <marcosmarxm@gmail.com>

avirajsingh7 committed:
🎉 New Source : Avni (Low Code Cdk) (#30222)

Signed-off-by: Aviraj Gour <100823015+avirajsingh7@users.noreply.github.com>
Co-authored-by: Siddhant Singh <ssiddhant3030@gmail.com>
Co-authored-by: Marcos Marx <marcosmarxm@users.noreply.github.com>
Co-authored-by: marcosmarxm <marcosmarxm@gmail.com>

nobowned committed:
✨ Destination Typesense: Added support for path config option (#36460)

Co-authored-by: Marcos Marx <marcosmarxm@users.noreply.github.com>
Co-authored-by: marcosmarxm <marcosmarxm@gmail.com>

stephane-airbyte committed:
fix ContainerFactory (#36574)

henriblancke committed:
🐛  Destination AWS Datalake: Update to support new state protocol (#36386)

Signed-off-by: Henri Blancke <blanckehenri@gmail.com>
Co-authored-by: Marcos Marx <marcosmarxm@users.noreply.github.com>
Co-authored-by: marcosmarxm <marcosmarxm@gmail.com>

gosusnp committed:
Update cloud record size limit doc (#36521)

Co-authored-by: Evan Tahler <evan@airbyte.io>

girarda committed:
Update connector-metadata-file.md to mention the icon field is deprecated (#36582)

girarda committed:
Python CDK: rename a unit test (#36556)

dependabot[bot] committed:
Bump black from 22.1.0 to 24.3.0 in /airbyte-cdk/python (#36568)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump requests from 2.25.1 to 2.31.0 in /airbyte-ci/connectors/common_utils (#36593)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

natikgadzhi committed:
connector-templates: Unpinning pytest and upgrading to recent base image version (#36596)

natikgadzhi committed:
airbyte-ci: Remove qa-engine in favor of connectors_qa (#36591)

maxi297 commented on airbyte-integrations/connectors/source-shopify/metadata.yaml:
```diff
@@ -17,6 +17,7 @@ data:
   githubIssueLabel: source-shopify
   icon: shopify.svg
   license: ELv2
+  maxSecondsBetweenMessages: 60
```
Checking shopify, which rate limit is by minute? I only see some per second

strosek replied:
You're right. I didn't double check the ones not included originally. I'll change it to 1s, and FB to 300s.

graingert-coef committed:
docs: fix pagination link header documentation (#36599)

natikgadzhi committed:
Security: resolve cryptography-related warnings (#36597)

tolik0 committed:
:sparkles: Source Microsoft SharePoint: Update QL to 400 (#36573)

gisripa committed:
Destination Postgres: `airbyte_meta`, CDK upgrade and other fixes (#36363)

stephane-airbyte committed:
disable spotbugs for CDK test and testFixtures submodules (#36606)

Some CDK submodules are failing their spotbugs checks. We're disabling those checks so we can publish the CDK

Commit:
Correct Facebook Marketing and Shopify maxSecsBetweenMessages

Commit:
Correct Facebook Marketing and Shopify maxSecsBetweenMessages

roman-yermilov-gl committed:
Source Mailchimp: fix acceptance tests (#36601)

stephane-airbyte committed:
disable some spotbugs checks for cdk tests (#36611)

Commit:
java-cdk: remove airbyte-api generation, pull dependency instead (#36610)

See https://github.com/airbytehq/airbyte/pull/36322#discussion_r1543361439

natikgadzhi committed:
ci_credentials: add cryptography back 🙈 (#36614)

Commit:
🎉 Source Zendesk Chat: migrate to `YamlDeclarativeSource (Low-code)` (#35867)

darynaishchenko committed:
:sparkles: Source Sentry: migrate to low code (#35755)

Commit:
🎉 Source Recharge: migrate to Low-Code (#35450)

natikgadzhi committed:
airbyte-ci: Update pipelines requests and pyjwt (#36607)

brianjlai committed:
🐛 [source-slack] bump slack to 0.77.2 to emit record counts as float (#36579)

lazebnyi committed:
follow up to #36294: allow migrate sub stream state with custom partition router (#36590)

lazebnyi committed:
🤖 Bump minor version of Python CDK

aaronsteers committed:
PyAirbyte: delete `airbyte-lib` folder (#35910)

Co-authored-by: Natik Gadzhi <natik@respawn.io>

Commit:
Merge branch 'strosek/issue-6779' of github.com:airbytehq/airbyte into strosek/issue-6779
